src/app/forms/custom-sections/tyres/tyres.component.ts

```diff
@@ -0,0 +1,93 @@
+import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output } from '@angular/core';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormArray, CustomFormGroup, FormNodeEditTypes } from '@forms/services/dynamic-form.types';
+import { getTyresSection } from '@forms/templates/general/tyres.template';
+import { Axle, TechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
+import { Subscription, debounceTime } from 'rxjs';
+
+@Component({
+  selector: 'app-tyres',
+  templateUrl: './tyres.component.html',
+  styleUrls: ['./tyres.component.scss']
+})
+export class TyresComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() vehicleTechRecord!: TechRecordModel;
+  @Input() isEditing = false;
+  public isError: boolean = false;
+  public errorMessage?: string;
+
+  @Output() formChange = new EventEmitter();
+
+  public form!: CustomFormGroup;
+  private _formSubscription = new Subscription();
+
+  constructor(public dfs: DynamicFormService) {}
+
+  ngOnInit(): void {
+    this.form = this.dfs.createForm(this.template, this.vehicleTechRecord) as CustomFormGroup;
+    this._formSubscription = this.form.cleanValueChanges.pipe(debounceTime(400)).subscribe(event => {
+      this.formChange.emit(event);
+    });
+  }
+
+  ngOnChanges() {
+    this.form?.patchValue(this.vehicleTechRecord, { emitEvent: false });
+  }
+
+  ngOnDestroy(): void {
+    this._formSubscription.unsubscribe();
+  }
+
+  get template() {
+    switch (this.vehicleTechRecord.vehicleType) {
+      case VehicleTypes.PSV:
+        return getTyresSection(VehicleTypes.PSV);
+      case VehicleTypes.HGV:
+        return getTyresSection(VehicleTypes.HGV);
+      case VehicleTypes.TRL:
+        return getTyresSection(VehicleTypes.TRL);
+    }
+  }
```

Comment:
😊
```suggestion
  get template() {
    return getTyresSection(this.vehicleTechRecord.vehicleType);
  }
```

Reply:
I'm leaving this here for your visibility, you can add it later when you do the tyres work.

Reply:
Lol, I wrote this whole section in 20 mins just to prove the axle buttons work, have to say this is quite a funny error from me haha! I'm actually going to leave it on this branch, I'll fix it on the correct branch for this work which I am starting on now. Ticket this sprint to cover